.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -307,6 +322,23 @@ jobs:
     - name: "Print compiler cache statistics"
       run: sccache --show-stats
 
+
+    # This is a workaround for the fact that GH doesn't support updating existing caches.
+    # See: https://github.com/azu/github-actions-overwrite-cache-example
+    - name: "Delete previous compiler cache"
+      if: ${{ steps.compiler-cache-restore.outputs.cache-hit }}
+      continue-on-error: true
+      run: |
+        gh extension install actions/gh-actions-cache
+        gh actions-cache delete "${{ matrix.name }}_compiler_cache" --confirm
```

Comment:
```console

Run gh extension install actions/gh-actions-cache
Error: failed to run git: fatal: not a git repository (or any of the parent directories): .git
. error: exit status 128
Usage:
  gh-actions-cache delete <key> [flags]

Flags:
  -B, --branch string   Filter by branch
      --confirm         Delete the cache without asking user for confirmation.
  -h, --help            help for delete
  -R, --repo string     Select another repository for finding actions cache.

Error: Process completed with exit code 1.
```

Reply:
Noticed that as well...
I'm a bit confused why that happens...
Maybe need to set the working_dir or something :thinking:

Reply:
```
  gh extension install actions/gh-actions-cache
  gh actions-cache delete "Ubuntu 22.04 (gcc) Qt5_compiler_cache" --confirm
  shell: /usr/bin/bash -e {0}
  env:
    MACOSX_DEPLOYMENT_TARGET: 12.7
    SCCACHE_CACHE_SIZE: 600M
    GH_TOKEN: ***
  
Error: Resource not accessible by integration
Error: Process completed with exit code 1.
```

Hm, this may be because it's a forked repository :thinking: 
As I'm adding the `permissions: write`...
We may have to test this on a branch within the main repo.
I'll try merging it into a branch there.

Reply:
Assume it's not related to the custom path we use from...
```
    # On Windows, the D drive that the workspace is on by default runs out of space when
    # vcpkg builds Qt, so the repo needs to be cloned to the C drive. GitHub's checkout
    # action does not support cloning to a path outside GITHUB_WORKSPACE:
    # https://github.com/actions/checkout/issues/197
    - name: "Clone Git repository"
      uses: LebedevRI/checkout@issue197
      with:
        path: ${{ matrix.workspace }}
```

Reply:
And on linux we use `workspace: /home/runner/cxx-qt`

Reply:
Ah I see you solved that and now it has permission issues